apps/aefate/src/aefa_fate.erl

```diff
@@ -0,0 +1,475 @@
+-module(aefa_fate).
+-export([run/2]).
+
+-export([get_trace/1]).
+
+-include("aefa_data.hrl").
+
+-ifdef(TEST).
+-define(trace(I,S), S#{trace => [{I, erlang:process_info(self(), reductions)} |get_trace(S)]}).
+-else.
+-define(trace(I,S), S).
+-endif.
+
+run(What, Chain) ->
+    EngineState = setup_engine(What, Chain),
+    execute(EngineState).
+
+execute(EngineState) ->
+    Instructions = current_bb(EngineState),
+    loop(Instructions, EngineState).
```

Comment:
It seems a bit error prone to have 2 copies of current_bb, one update in the loop in EngineState and one updated in the recurser...

Reply:
Yes. Not so happy with that. Ill try to fix it.